Source/WebCore/css/CSSFontFace.h

```diff
@@ -84,22 +87,18 @@ class CSSFontFace final : public RefCounted<CSSFontFace> {
     //              \/  \/    \/  \/
     //             Success    Failure
     enum class Status : uint8_t { Pending, Loading, TimedOut, Success, Failure };
-    
+
     struct UnicodeRange;
-    
-    // Optional return values to represent default string for members of FontFace.h
-    const std::optional<CSSValueList*> families() const { return m_status == Status::Failure ? std::nullopt : static_cast<std::optional<CSSValueList*>>(m_families.get()); }
-    std::optional<FontSelectionRange> weight() const { return m_status == Status::Failure ? std::nullopt : static_cast<std::optional<FontSelectionRange>>(m_fontSelectionCapabilities.computeWeight()); }
-    std::optional<FontSelectionRange> stretch() const { return m_status == Status::Failure ? std::nullopt : static_cast<std::optional<FontSelectionRange>>(m_fontSelectionCapabilities.computeWidth()); }
-    std::optional<FontSelectionRange> italic() const { return m_status == Status::Failure ? std::nullopt : static_cast<std::optional<FontSelectionRange>>(m_fontSelectionCapabilities.computeSlope()); }
-    std::optional<FontSelectionCapabilities> fontSelectionCapabilities() const { return m_status == Status::Failure ? std::nullopt : static_cast<std::optional<FontSelectionCapabilities>>(m_fontSelectionCapabilities.computeFontSelectionCapabilities()); }
-    const std::optional<Vector<UnicodeRange>> ranges() const { return m_status == Status::Failure ? std::nullopt : static_cast<std::optional<Vector<UnicodeRange>>>(m_ranges); }
-    const std::optional<FontFeatureSettings> featureSettings() const { return m_status == Status::Failure ? std::nullopt : static_cast<std::optional<FontFeatureSettings>>(m_featureSettings); }
-    std::optional<FontLoadingBehavior> loadingBehavior() const { return m_status == Status::Failure ? std::nullopt :  static_cast<std::optional<FontLoadingBehavior>>(m_loadingBehavior); }
-    void setWeight(FontSelectionRange weight) { m_fontSelectionCapabilities.weight = weight; }
-    void setStretch(FontSelectionRange stretch) { m_fontSelectionCapabilities.width = stretch; }
-    void setStyle(FontSelectionRange italic) { m_fontSelectionCapabilities.slope = italic; }
+
+    RefPtr<CSSValueList> families() const;
+    Span<const UnicodeRange> ranges() const { ASSERT(m_status != Status::Failure); return m_ranges.span(); }
+
+    // FIXME: Consider adding the word "compute" to this function's name.
+    FontSelectionCapabilities fontSelectionCapabilities() const { ASSERT(m_status != Status::Failure); return m_fontSelectionCapabilities.computeFontSelectionCapabilities(); }
+
+    // FIXME: Consider adding the word "specified" to this function's name.
     void setFontSelectionCapabilities(FontSelectionCapabilities capabilities) { m_fontSelectionCapabilities = capabilities; }
```

Comment:
Could you file a bug for these FIXME?

Reply:
I’d rather take them out. I don’t think this ideas are so good that they should have bug reports.